### DISCUSSION TITLE
Use environment variables or secrets in job container image

### DISCUSSION THREAD

**Discussion by ashrafguitoni** [community/community#30505] [Actions] [answered]
It's currently not possible to run a GHA job inside an image with a dynamic name:
```python
jobs:
  my-job:
    container:
      image: ${{ secrets.DOCKER_REGISTRY_NAME }}/my-image:${{ github.sha }}
      credentials:
        username: ${{ secrets.DOCKER_REGISTRY_USERNAME }}
        password: ${{ secrets.DOCKER_REGISTRY_PASSWORD }}
```

If I try to run something like this I get:
```bash
Unrecognized named-value: 'secrets'. Located at position 1 within expression: secrets.DOCKER_REGISTRY_NAME
```

Although I can use `${{ secrets.DOCKER_REGISTRY_NAME }}` in docker build jobs, I can't use them to pick an image for other jobs to run inside. I think it would be really great to have!

**Comment by Akash1134** [comment, marked as answer]
Hi @ashrafguitoni This could be a great feature to have. I have labelled the discussion with the relevant tag so our engineers can take a look. Thanks!

**Reply by ashrafguitoni** [reply]
Thank you! I'm looking forward to having this implemented.

**Reply by BeyondEvil** [reply]
Please add support for the `env` context as well! 🙏

**Comment by BeyondEvil** [comment]
I want this too. Honestly I was quite surprised that this wasn't already supported for `secrets` and `env`!

**Comment by junchen6072** [comment]
I would also love to have this!
